### DISCUSSION TITLE
Developers need to be able to set a canister ID for a canister on the IC

### DISCUSSION THREAD

**Discussion by hansl** [dfinity/sdk#2646] [Feature Requests]
Developers might create canisters through the Wallet UI, and would want to install their WASM using `dfx`. Something like `dfx canister set-canister-id hello aaaaa-aa` would probably be enough for this use case.

**Comment by lsgunnlsgunn** [comment]
We need to capture some details about this feature...
What is the intended user scenario/use case for the create a new canister from the wallet UI? 
If the operation is successful it reserves a new identifier, but then what? 
It isn't really part of any project or have any code associated with it, so how do I interact with it? 
Where does the canister "live".? How do I put things in it? Where is the new id recorded?
